app/notification_email.go

```diff
@@ -284,6 +288,23 @@ func getFormattedPostTime(user *model.User, post *model.Post, useMilitaryTime bo
 	}
 }
 
+func (a *App) generateHyperlinkForChannels(postMessage, teamName, teamURL string) string {
+	r := regexp.MustCompile(`\~[-a-zA-Z0-9]+\b`)
+	matches := r.FindAllString(postMessage, -1)
+	var channelURL, channelHyperLink string
+	for _, channelName := range matches {
+		ch, err := a.GetChannelByNameForTeamName(channelName[1:], teamName, true)
+		if ch == nil || err != nil {
+			mlog.Warn(fmt.Sprintf("Encountered error while generating hyperlink for %s, as channel with that name doesn't exist.", channelName[1:]))
```

Comment:
Can you downgrade this to an `mlog.Debug`? This is something that can happen normally if a user mistypes a channel name or uses one of the Markdown things I mentioned above, so I don't think it needs to be a warning message that might clutter up the logs.